File: MyWatchlist/MyWatchlistSpring/src/main/java/it/niko/mywatchlist/entities/Watchlist.java
package it.niko.mywatchlist.entities;

import lombok.Getter;
import lombok.Setter;
import javax.persistence.*;

@Getter
@Setter
@Entity(name = "Watchlist")
@Table(name = "watchlist", schema = "mywatchlist")
public class Watchlist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "series_id")
    private Series series;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "status_id")
    private Status status;

    @Column(name = "progress")
    private int progress;

    @Column(name = "score")
    private int score;

    @Column(name = "comment")
    private String comment;
}

File: MyWatchlist/MyWatchlistAngular/src/app/models/watchlist.ts
import { Status } from "./status";
import { Series } from "./series";

export class Watchlist {
  id: number;
  series: Series;
  status: Status;
  progress: number;
  comment: string;
  score: number;

  constructor(id:number, series: Series, status: Status, progress: number,
              comment: string, score: number) {
    this.id = id; this.series = series;
    this.status = status; this.progress = progress;
    this.comment = comment; this.score = score;
  }
}
